src/Microsoft.VisualStudio.Editor.Razor/RazorDirectiveCompletionSourceProvider.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    [System.Composition.Shared]
+    [Export(typeof(IAsyncCompletionSourceProvider))]
+    [Name("Razor directive completion provider.")]
+    [ContentType(RazorLanguage.ContentType)]
```

Comment:
Should this be the CORE content type if this is for 15.8?

Reply:
It will be once VS Editor + Roslyn move to VS 15.8. For now these completion changes require private builds of their bits which target 15.7 😢